Katello/katello
# Fixes #33254 - CVV errata count wrong with date filter and when errata packages aren't in the source repo

```ruby
if erratum.packages.any? && (erratum.packages.pluck(:filename) - rpm_filenames).empty?
```
made the assumption that, if an erratum has any package filenames that aren't in the source repository, it shouldn't be included in the content view version. This caused low errata counts because errata can include RPMs that aren't in the source repository.

This issue occurs when multiple repositories share an erratum.  One repo could have the erratum and 2 of its 3 packages.  The other repo could have the 1 other package.  Neither repo has all the packages, so the erratum is incorrectly excluded from the content view version.  I fixed this by doing an intersection (`&`) instead of subtraction.

Also, as a secondary issue, the date filters were being mostly ignored.

2 things to test:

First:

1) Sync "Red Hat Enterprise Linux 7 Server - Update Services for SAP Solutions - Optional RPMs x86_64 7.7" and "Red Hat Enterprise Linux 7 Server - Update Services for SAP Solutions RPMs x86_64 7.7".
2) Stick them in a content view. Add an inclusion filter that should include most of the errata (like a date filter).
3) Publish the new content view version. See that many errata are missing (thousands vs hundreds).
4) Patch in my PR and publish again. There should be many more errata now.
5) Check that other content view publishing with filters makes sense.

Second:

1) Sync a repo with errata but no modules (like https://jlsherrill.fedorapeople.org/fake-repos/needed-errata/).
2) Add that repo to a content view.
3) Create a date filter so that only a couple errata are included.
4) Publish, and see that the date filter is ignored.
5) Patch in my PR and publish again. The date filter should be obeyed.
6) Try the same thing with an exclusion filter.
7) Do the same thing with a repo that has errata and modules (like https://fixtures.pulpproject.org/rpm-with-modules/)

Perhaps try testing the first and second cases together too to be safe.
